utils/s2n_init.c

```diff
@@ -89,12 +95,16 @@ static bool s2n_cleanup_atexit_impl(void)
     /* the configs need to be wiped before resetting the memory callbacks */
     s2n_wipe_static_configs();
 
-    return s2n_result_is_ok(s2n_cipher_suites_cleanup()) &&
+    bool cleaned_up = \
+        s2n_result_is_ok(s2n_cipher_suites_cleanup()) &&
         s2n_result_is_ok(s2n_rand_cleanup_thread()) &&
         s2n_result_is_ok(s2n_rand_cleanup()) &&
         s2n_result_is_ok(s2n_libcrypto_cleanup()) &&
         s2n_result_is_ok(s2n_locking_cleanup()) &&
         (s2n_mem_cleanup() == S2N_SUCCESS);
+
+    initialized = !cleaned_up;
```

Comment:
Why leave initialized==true if cleanup fails?

If it's so that we can retry cleanup, isn't that a problem if some of the cleanup methods aren't idempotent? Or is the idea that the next call to s2n_init should fail?

Reply:
Correct; we did not fully clean up, so we are still initialized.  Calling s2n_init in this state may leak memory or have other side-effects.

Reply:
Got it. And being able to call s2n_cleanup again isn't as big of a problem, because worst case the call fails.